protocol-spec/enumeration.md

```diff
@@ -127,6 +135,61 @@ None. Server-to-Client only.
 
 **Description:** Server reply to a client request for a device list.
 
+**Introduced In Version:** 0
+
+**Message Version:** 1
+
+**Fields:**
+
+* _Id_ \(unsigned int\): Message Id
+* _Devices_ \(array\): Array of device objects
+  * _DeviceName_ \(string\): Descriptive name of the device
+  * _DeviceIndex_ \(unsigned integer\): Index used to identify the device when sending Device Messages.
+  * _DeviceMessages_ \(dictionary\): Accepted Device Messages 
+    * Keys \(string\): Type names of Device Messages that the device will accept
+    * Values \(dictionary\): Attributes for the Device Messages. Both keys and values are strings.
```

Comment:
Why do they have to be string -> string dicts? We're spec'ing everything that can go in here, these should be the types we spec. We shouldn't expect the user to have to discern the types themselves.

Reply:
This was for schema flexibility: new properties can be added without breaking backwards compatibility. The values are also strings in order to avoid overly complex changes to the parser: many JSON parsers will refuse to handle mixed type maps, or will require explicit object mappings (which would defeat the point of making the field flexible).

We could define each property and make them explicit, adding property stripping into the message version downgrade code too. I'm leaning towards this option more and more.